### PR TITLE
Map the in-cluster context to default cluster alias when reconciling the Tekton pipeline runs

### PR DESCRIPTION
**Pull request by ccojocar**
Follow up #12177

### CODE REVIEW

File: prow/pjutil/pjutil.go [hunk 1]
```diff
@@ -278,3 +279,11 @@ func JobURL(plank config.Plank, pj prowapi.ProwJob, log *logrus.Entry) string {
 	}
 	return ""
 }
+
+// ClusterToCtx converts the prow job's cluster to a cluster context
+func ClusterToCtx(cluster string) string {
+	if cluster == kube.InClusterContext {
+		return kube.DefaultClusterAlias
```
Comment (stevekuznetsov) [inline prow/pjutil/pjutil.go]: Do we validate that two aliases not point to the same place when someone provides explicit config?

File: prow/pjutil/pjutil.go [hunk 2]
```diff
@@ -278,3 +279,11 @@ func JobURL(plank config.Plank, pj prowapi.ProwJob, log *logrus.Entry) string {
 	}
 	return ""
 }
+
+// ClusterToCtx converts the prow job's cluster to a cluster context
+func ClusterToCtx(cluster string) string {
```
Comment (fejta) [inline prow/pjutil/pjutil.go]: Not 100% convinced I see the value in exposing this two line method.

File: prow/pjutil/pjutil.go (hunk 1 above)
Comment (fejta) [reply prow/pjutil/pjutil.go]: Not right now

File: prow/pjutil/pjutil.go (hunk 2 above)
Comment (ccojocar) [reply prow/pjutil/pjutil.go]: Should we keep it local in each controller?

Comment (fejta) [reply prow/pjutil/pjutil.go]: Don't feel super strongly about it one way or another.

Comment (ccojocar) [reply prow/pjutil/pjutil.go]: Alright, then let's reuse it like this.

Comment (fejta) [reply prow/pjutil/pjutil.go]: Give it a `/hold cancel` when you're ready for it to merge!

Comment (ccojocar) [reply prow/pjutil/pjutil.go]: I validated it today, and found a small issue. The local context needs to be set before truncating the config map. 

I will send this fix in a separate PR also for build controller. 

This PR is ready for merging.

Comment (ccojocar) [reply prow/pjutil/pjutil.go]: https://github.com/kubernetes/test-infra/pull/12320

**Pull request merged**